traitsui/testing/tester/registry_helper.py

```diff
@@ -12,37 +12,71 @@
 from traitsui.testing.tester import locator
 
 
-def find_by_id_in_nested_ui(wrapper, location):
-    """ Helper function for resolving from a target to a TargetById. The
-    target must have a solver registered from it to an instance of
-    traitsui.ui.UI
+def _get_editor_by_name(ui, name):
+    """ Return a single Editor from an instance of traitsui.ui.UI with
+    a given extended name. Raise if zero or many editors are found.
 
     Parameters
     ----------
-    wrapper : UIWrapper
-    location : instance of locator.TargetById
+    ui : traitsui.ui.UI
+        The UI from which an editor will be retrieved.
+    name : str
+        A single name for retrieving an editor on a UI.
+
+    Returns
+    -------
+    editor : Editor
+        The single editor found.
     """
-    new_interactor = wrapper.locate(locator.NestedUI())
-    return new_interactor.find_by_id(location.id).target
+    editors = ui.get_editors(name)
+
+    all_names = [editor.name for editor in ui._editors]
+    if not editors:
+        raise ValueError(
+            "No editors can be found with name {!r}. "
+            "Found these: {!r}".format(name, all_names)
+        )
+    if len(editors) > 1:
+        raise ValueError(
+            "Found multiple editors with name {!r}.".format(name))
+    editor, = editors
+    return editor
 
 
-def find_by_name_in_nested_ui(wrapper, location):
-    """ Helper function for resolving from a target to a TargetByName. The
-    target must have a solver registered from it to an instance of
-    traitsui.ui.UI
+def _get_editor_by_id(ui, id):
+    """ Return single Editor from an instance of traitsui.ui.UI with
+    the given identifier.
 
     Parameters
     ----------
-    wrapper : UIWrapper
-    location : instance of locator.TargetByName
+    ui : traitsui.ui.UI
+        The UI from which an editor will be retrieved.
+    id : str
+        Id for finding an item in the UI.
+
+    Returns
+    -------
+    editor : Editor
+        The single editor found.
     """
-    new_interactor = wrapper.locate(locator.NestedUI())
-    return new_interactor.find_by_name(location.name).target
+    try:
+        editor = getattr(ui.info, id)
+    except AttributeError:
+        raise ValueError(
+            "No editors found with id {!r}. Got these: {!r}".format(
+                id, ui._names)
+            )
+    return editor
 
 
 def register_nested_ui_solvers(registry, target_class, nested_ui_getter):
-    """ Function to register solvers for a particular target type to
-    NestedUIs and TargetByNames within those NestedUIs.
+    """ Function to register solvers for obtaining nested targets inside a
+    nested traitsui.ui.UI inside a (parent) target.
+
+    For example, a TreeEditor may contain a nested UI from which more nested
+    UI target may be required. In that case, the ``target_class`` will the
+    TreeEditor and the ``nested_ui_getter`` specifies how to obtain the nested
+    UI from it.
```

Comment:
This makes sense, but was a bit confusing to read at first. 
Specifically "from which more nested UI target may be required."  You are basically saying that the tree editor may have a Nested UI call it newUI, and then newUI may have more nested UIs itself.
That is a possibility, but there is also the possibility that the tree editor has a nested UI, newUI, and newUI just contains say a button for example to interact with.  But then the next line starts with "In that case", which to me at first read as only in the case where newUI has more nested UIs.

Feel free to ignore this as at this point what you have written makes perfect sense to me.  It just took me a second to process initially, but that very well may have just been me.

Reply:
Thanks! Indeed, the word UI is overloaded here which causes confusion. I will try to fix that.